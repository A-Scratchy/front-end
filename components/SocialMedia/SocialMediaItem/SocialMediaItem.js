import React from 'react';
import PropTypes from 'prop-types';
import OutboundLink from 'components/_common_/OutboundLink/OutboundLink';
import { withRouter } from 'next/router';
import styles from './SocialMediaItem.css';

SocialMediaItem.propTypes = {
  name: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  svg: PropTypes.node.isRequired,
};

function SocialMediaItem(props) {
  const { name, href, svg } = props;
  return (
    <div className={styles.SocialMediaItem}>
      <OutboundLink
        analyticsEventLabel={`Social Media Engagement: ${name}`}
        hasIcon={false}
        href={href}
        router={props}
      >
        {svg}
      </OutboundLink>
    </div>
  );
}

export default withRouter(SocialMediaItem);
