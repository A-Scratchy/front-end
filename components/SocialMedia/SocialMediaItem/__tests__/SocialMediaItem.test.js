import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import SocialMediaItem from '../SocialMediaItem';

describe('SocialMediaItem', () => {
  test('it should render properly with required (all) props', () => {
    createSnapshotTest(
      <SocialMediaItem
        href="https://facebook.com"
        imageSource="/static/images/icons/facebook_logo.svg"
        alt="Facebook"
      />,
    );
  });
});
