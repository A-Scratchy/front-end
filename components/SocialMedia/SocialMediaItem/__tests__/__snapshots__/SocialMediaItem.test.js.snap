// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaItem should render with required props 1`] = `
<div
  className="SocialMediaItem"
>
  <a
    className=""
    href="https://facebook.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="screenReaderOnly"
    >
      Opens in new window
    </span>
    <test-file-mock
      fill="blue"
    />
  </a>
</div>
`;
